main.go

```diff
@@ -214,12 +221,93 @@ func defaultHandler(w http.ResponseWriter, r *http.Request) {
 
 	log.Printf("new request %s %s", r.Method, r.RequestURI)
 
+	// there are two ways that a request gets hyjacked:
+	// 1 - X-Return-* header
+	// 2 - Config
+	// An X-Return-* header always overrides config.
+	requestHyjacked := false
+	var delay time.Duration
+	var code int
+	var headers http.Header
+	var data []byte
+	var err error
+
+	if hdr := r.Header.Get("X-Return-Delay"); hdr != "" {
+		delay, err = time.ParseDuration(hdr)
+		if err != nil {
+			log.Println("cannot set delay", err)
```

Comment:
should we return an error here? or is it acceptable to just not set a delay if we fail to parse the delay?

Reply:
talked offline; the utility is run in a testing situation and does not need robust error handling and logging should be enough